bin/schema_generator.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# TODO: Update schema mapping for validation
+# TODO: Handle overwriting glean schemas
+# TODO: Include Main Ping from schema generation
+# TODO: What the heck to do with pioneer-study, a non-nested namespace
+
+if [[ -z $MOZILLA_PIPELINE_SCHEMAS_SECRET_GIT_SSHKEY ]]; then
+    echo "Missing secret key" 1>&2
+    exit 1
+fi
+
+mkdir -p /app/.ssh
+echo $MOZILLA_PIPELINE_SCHEMAS_SECRET_GIT_SSHKEY > /app/.ssh/id_rsa
+
+chmod 700 "$HOME/.ssh"
+chmod 700 "$HOME/.ssh/id_rsa"
+
+DEV_BRANCH="dev" # Branch we'll work on
+MPS_BRANCH="generated-schemas" # Branch we'll push to
+
+# 0. Install dependencies
+
+cargo install jsonschema-transpiler
+
+virtualenv mgs-venv 
+source mgs-venv/bin/activate
+pip install -U mozilla-schema-generator
+
+# 1. Pull in all schemas from MPS
+
+SCHEMAS_DIR="schemas"
+BASE_DIR="/app"
+
+cd $BASE_DIR
+rm -rf mozilla-pipeline-schemas
+
+git clone https://www.github.com/mozilla-services/mozilla-pipeline-schemas.git
+cd mozilla-pipeline-schemas/$SCHEMAS_DIR
+git checkout $DEV_BRANCH
+
+# 2. Remove all non-json schemas (e.g. parquet)
+
+find . -not -name "*.schema.json" -type f | xargs rm
+
+# 3. Generate new schemas
+
+mozilla-schema-generator generate-glean-ping --out-dir . --pretty
+mozilla-schema-generator generate-main-ping --out-dir ./telemetry --pretty --split
+
+# 3a. Keep only whitelisted schemas
```

Comment:
"blacklist" and "whitelist" are going the way of "slave"/"master" terminology. If we can avoid them, I think we should. I've found "allowlist" and "blocklist" to be suitable alternatives.

Reply:
Good catch - I had never realized this. Updated!